lesson24-Home-Work38/script/weather.js

```diff
@@ -0,0 +1,95 @@
+const api = {
+    key: "&units=metric&APPID=5d066958a60d315387d9492393935c19",
+    base: "http://api.openweathermap.org/data/2.5/weather?q="
+};
+
+const iconURL = "http://openweathermap.org/img/w/";
+
+const searchBtn = document.querySelector(".search-city-btn");
+const cityValue = document.querySelector(".search-box");
+const widget = document.querySelector(".container");
+const city = document.querySelector(".city");
+const date = document.querySelector(".date");
+const temperature = document.querySelector(".temp");
+const pressure = document.querySelector(".pressure");
+const weatherDescr = document.querySelector(".weather-description");
+const humidity = document.querySelector(".humidity");
+const speedWind = document.querySelector(".speed");
+const degWind = document.querySelector(".deg");
+const weatherPict = document.querySelector(".icon");
+const highLowDiv = document.querySelector(".hi-low");
+const error = document.querySelector(".error");
+const form = document.querySelector(".form");
+
+function getData(cityName) {
+    fetch(`${api.base}${cityName}${api.key}`)
+        .then(response => response.json())
+        .then(data => {
+            showData(data);
+            cityValue.value = "";
+            cityValue.focus();
+        })
+        .catch(() => {
+            error.innerHTML = "You enter not valid city. Please, enter again";
+            cityValue.value = "";
+            showElement(error);
+            hideElement(widget);
+            cityValue.focus();
+        });
+}
+
+function showData(data) {
+    city.innerHTML = `${data.name}, ${data.sys.country}`;
```

Comment:
Why not try the template approach?

```
function showData(data) {
  const element = document.querySelector(". main");
  
  element.innerHtml = (
  `
    <div class="container hide">
      <section class="location">
        <div class="city">${data.name}, ${data.sys.country}</div>
        <div class="date">${showDate()}</div>
      </section>
  
      <div class="current">
        // the same here
      </div>
    </div>
  `
  );
}
```

Then you will have less list of selectors at the top of the file
```
const city = document.querySelector(".city");
const date = document.querySelector(".date");
const temperature = document.querySelector(".temp");
const pressure = document.querySelector(".pressure");
/// ...
```

Reply:
To reset when you fetch the data you just need

```
element.innerHtml = "";
```